backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -2068,27 +2069,51 @@ public void logVersionedInput(LogVersionedInput request)
       if (existingMappings.isEmpty()) {
         existingMappings.addAll(versioningModeldbEntityMappings);
       } else {
-        List<VersioningModeldbEntityMapping> finalVersionList = new ArrayList<>();
-        for (VersioningModeldbEntityMapping versioningModeldbEntityMapping :
-            versioningModeldbEntityMappings) {
-          boolean addNew = true;
-          for (VersioningModeldbEntityMapping existsVerMapping : existingMappings) {
-            if (versioningModeldbEntityMapping.equals(existsVerMapping)) {
-              addNew = false;
-              break;
+        if (!versioningModeldbEntityMappings.isEmpty()) {
+          VersioningModeldbEntityMapping existingFirstEntityMapping = existingMappings.get(0);
+          VersioningModeldbEntityMapping versioningModeldbFirstEntityMapping =
+              versioningModeldbEntityMappings.get(0);
+          if (!existingFirstEntityMapping
+                  .getRepository_id()
+                  .equals(versioningModeldbFirstEntityMapping.getRepository_id())
+              || !existingFirstEntityMapping
+                  .getCommit()
+                  .equals(versioningModeldbFirstEntityMapping.getCommit())) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaDelete<VersioningModeldbEntityMapping> delete =
+                cb.createCriteriaDelete(VersioningModeldbEntityMapping.class);
+            Root<VersioningModeldbEntityMapping> e =
+                delete.from(VersioningModeldbEntityMapping.class);
+            delete.where(cb.in(e.get("experimentRunEntity")).value(runEntity));
+            Transaction transaction = session.beginTransaction();
+            session.createQuery(delete).executeUpdate();
+            transaction.commit();
```

Comment:
this code is equivalent to deleting all current mappings of experiment run.
I haven't used for each delete here because hibernate translate delete statement to sql wrong if there are null index for VersioningModeldbEntityMapping.

Reply:
I believe the ticket and @convoliution are expecting a conflict when there is versionMap already logged.

Reply:
I see that there are no comments in this conversation: https://vertaai.slack.com/archives/CJG0NL86B/p1588805741019200 after a comment about about overwrite. Was it agreed? @conradoverta

Reply:
added a code to raise conflict exception